src/ccnl-utils/src/ccn-lite-ctrl.c

```diff
@@ -1627,11 +1628,21 @@ main(int argc, char *argv[])
 #endif
             break;
         case 'u':
-            udp = strdup(optarg);
-            udp = strtok(udp, "/");
+            udp_temp = strdup(optarg);
+            /** strdup failed to allocate memory */
+            if (!udp_temp) {
+                return -1;
+            }
+
+            udp = strtok(udp_temp, "/");
             port = strtol(strtok(NULL, "/"), NULL, 0);
             use_udp = 1;
             printf("udp: <%s> <%i>\n", udp, port);
+
+            /** if memory has been allocated previously, free it */
+            if (udp_temp) {
```

Comment:
Isn't `udp_temp` always non-`NULL` here? Is this check necessary?

Reply:
If Malloc returns NULL in strdup because system is out of memory it can be NULL

Reply:
But there is a check for that only 9 lines above ..

Reply:
Right. Missed that. So, we can remove that.